### DISCUSSION TITLE
MSAL and Quasar v2

### DISCUSSION THREAD

**Discussion by nstratton-lfa** [quasarframework/quasar#10334] [General - Components / Directives / etc]
Has anyone incorporated msal.PublicClientApplication, with the MS Popup login, into Quasar?

Not quite sure how the pieces fit together in boot, the popup and routing.

**Comment by ontwikkelfabriek** [comment]
@nstratton-lfa 

Maybe this will help:
https://devblogs.microsoft.com/azure-sdk/vue-js-user-authentication/

**Reply by BryanGilomen** [reply]
Anyone have any good examples of Quasar and MSAL? It is still a bit confusing on how to get a Quasar and MSAL to work together for authentication.

**Reply by nstratton-lfa** [reply]
I hear you. I didn't have time and left it for now. Maybe Auth0 would be an alternative, they've just released MSAL. The most recent Quasar survey asked what features need to be added. Auth was in the list and I made it my number one request.

**Reply by hawkeye64** [reply]
@ldiebold is working on an Auth extension for Quasar

**Reply by nstratton-lfa** [reply]
That is excellent news. He's a great advocate for Quasar and auth is a critical component. Work wouldn't touch the work I had done without auth and I didn't have the time to learn and implement it.

**Reply by ldiebold** [reply]
I had a quick look at the docs, and this doesn't look too difficult to implement. I have some experience with 0Auth2 and PKCE and it looks like MSAL does most of the work for you.

What are you using on the backend? .NET?

**Reply by nstratton-lfa** [reply]
I was using node.js but implementing auth in the MS space is too hard, there's very limited documentation and examples. Moving back end to c# is probably the best option. We're non-profit and have recently seen Auth0 may help. Love your Quasar videos. #goldcoastqld

**Reply by ldiebold** [reply]
Oh, you're on the Gold Coast in Queensland!? I literally just spoke to someone on the phone from the Gold Coast about Quasar about a week ago!

My girlfriend and I are currently trying to move to Brisbane!

As for Auth with Microsoft, I know how you feel. We wanted to create a teams app using Quasar... too much work to justify it! Hahaha

**Reply by alexandergroup-jnelson** [reply]
Nigel, I came across your post while trying to do the same thing while learning Quasar.  I ended up using @azure/msal-browser and learning about boot files.  I created the boot file below and in my App.vue I imported the msal and use it to call the Azure loginPopup function.  Works though I'm sure that it's not the best way.

/src/boot/VueMsal.js
import { boot } from 'quasar/wrappers'
import { PublicClientApplication } from '@azure/msal-browser'

let msal = {}
export default boot(async ({ app }) => {
  msal = new PublicClientApplication({
    auth: {
      clientId: '9c0327e9-xxxx-xxxx-xxxx-f8fe2c19ecc2',
      authority: 'https://login.microsoftonline.com/19a56f76-xxxx-xxxx-xxxx-e058e2c895f0',
      redirectUri: window.location.origin + window.location.pathname,
    },
  })
  msal = app.config.globalProperties.$msalInstance
  console.log(msal)
})
export { msal }

App.vue
...
import { msal } from './boot/VueMsal'
...
  setup () {
    onMounted(() => {
    msal.loginPopup({})
      .then(() => {
        console.log(msal.getAllAccounts()[0])
      })
    })
  }
